### ISSUE TITLE
actionlint inconsistently catching unclosed expressions

### ISSUE THREAD

**Issue by Vinesse** [rhysd/actionlint#169]
Actionlint is inconsistently catching varieties of "unclosed expressions". Examples below.

### ✅ Example 1 [missing closing braces]:

**Syntax:**
`CCI_TOKEN: ${{ secrets.CCI_TOKEN`

**Actionlint Output:**
```
.github/workflows [ actionlint -shellcheck= example-file.yaml                                                                               
example-file.yaml:12:35: unexpected EOF while lexing expression [expression]
   |
12 |   CCI_TOKEN: ${{ secrets.CCI_TOKEN             
   |                                   ^
.github/workflows [
```
This was successfully caught as expected.

### ❌ Example 2 [missing starting braces]

**Syntax:**
`CCI_TOKEN: $ secrets.CCI_TOKEN }}`

**Actionlint Output:**
`.github/workflows [ actionlint -shellcheck= example-file.yaml`

No output. Does not catch this.

### ❌ Example 3 [incomplete expression]

`CCI_TOKEN: secrets.CCI_TOKEN }}`

**Actionlint Output:**
`.github/workflows [ actionlint -shellcheck= example-file.yaml`
 
No output. Does not catch this.

---

Note: The vanilla github ci (without actionlint) does fail on example 2 & 3: `(Line: 37, Col: 27): The expression is not closed. An unescaped ${{ sequence was found, but the closing }} sequence was not found.` It would therefore be beneficial to make it so actionlint accounts for these variations of unclosed expressions as well.

**Comment by rhysd** [comment]
> (Line: 37, Col: 27): The expression is not closed. An unescaped ${{ sequence was found, but the closing }} sequence was not found.

Is this error message from GitHub correct? There is no `${{` in the input, but the error message says '`${{` was found'. It seems GitHub's behavior is not correct. I'm not sure actionlint should follow this behavior.
